commons-impl/src/main/java/com/connexta/transformation/commons/inmemory/status/InMemoryMetadataTransformation.java

```diff
@@ -0,0 +1,178 @@
+    this.startTime = Instant.now();
+    this.state = State.IN_PROGRESS;
+    this.metadataType = metadataType;
+    this.transformId = transformId;
+    this.requestInfo = requestInfo;
+  }
+
+  @Override
+  public String getMetadataType() {
+    return metadataType;
+  }
+
+  @Override
+  public Optional<InputStream> getContent() {
+    if (content == null) {
+      return Optional.empty();
+    } else {
+      try {
+        return Optional.of(ByteSource.wrap(content).openStream());
+      } catch (IOException e) {
+        LOGGER.error(
+            "Unable to read contents of [{}] metadata for transformation [{}].",
+            metadataType,
+            transformId);
+        return Optional.empty();
+      }
+    }
+  }
+
+  @Override
+  public Optional<String> getContentType() {
+    return Optional.ofNullable(contentType);
+  }
+
+  @Override
+  public OptionalLong getContentLength() {
+    if (content == null) {
+      return OptionalLong.empty();
+    } else {
+      return OptionalLong.of(content.length);
+    }
+  }
+
+  @Override
+  public void succeed(String contentType, InputStream contentStream)
+      throws TransformationException, IOException, IllegalStateException {
+    Instant completionTime = Instant.now();
+    synchronized (stateLock) {
+      checkForCompletion();
+      try {
+        content = ByteStreams.toByteArray(contentStream);
+      } finally {
+        try {
+          contentStream.close();
+        } catch (IOException e) {
+          LOGGER.debug(
+              "Unable to close contents stream of [{}] metadata for transformation [{}].",
+              metadataType,
+              transformId);
+        }
+      }
+      this.state = State.SUCCESSFUL;
+    }
+    this.contentType = contentType;
+    this.completionTime = completionTime;
+  }
+
+  @Override
+  public void fail(ErrorCode reason, String message)
+      throws TransformationException, IllegalStateException {
+    Instant completionTime = Instant.now();
+    synchronized (stateLock) {
+      checkForCompletion();
+      state = State.FAILED;
+    }
+    failureReason = reason;
+    failureMessage = message;
+    this.completionTime = completionTime;
+  }
+
+  @Override
+  public Optional<ErrorCode> getFailureReason() {
+    return Optional.ofNullable(failureReason);
+  }
+
+  @Override
+  public Optional<String> getFailureMessage() {
+    return Optional.ofNullable(failureMessage);
+  }
+
+  @Override
+  public String getTransformId() {
+    return transformId;
+  }
+
+  @Override
+  public RequestInfo getRequestInfo() {
+    return requestInfo;
+  }
+
+  @Override
+  public Instant getStartTime() {
+    return startTime;
+  }
+
+  @Override
+  public Optional<Instant> getCompletionTime() {
+    return Optional.ofNullable(completionTime);
+  }
+
+  @Override
+  public Duration getDuration() {
+    return Duration.between(startTime, getCompletionTime().orElse(Instant.now()));
```

Comment:
```suggestion
    return Duration.between(startTime, getCompletionTime().orElseGet(Instant::now));
```

This avoid calling `now()` if not required. Also maybe we should move this default implementation to the interface